integrationtests/mayavi/common.py

```diff
@@ -383,12 +386,18 @@ def run_standalone(self):
             type, value, tb = self.exception_info
             raise type, value, tb
 
+
     def run(self):
         """This starts everything up and runs the test.  Call main to
         run the test."""
         # Calls the users test code.
         try:
-            self.do()
+            if self.profile:
+                memory_assistant = MemoryAssistant()
+                memory_assistant.assertReturnsMemory(self.do_profile,
```

Comment:
As I mentioned earlier, `do` does a lot more stuff.

As per your earlier comment (quoting only for reference):
--------
It is not easy to understand what the test is actually testing. Do we really need to perform all these operations (e.g. load/save, manipulate contrours) to flesh out this specific memory leak? I would also expect that only the memory assertions are necessary.
--------

Hence a separate do_profile method. Separating is also not a great idea as then we have to classify between integration/memory tests. 

Alternative approach I can think of is: to refactor `do` into those separate methods and ignore those which are not relevant to memory checks.